src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6435,7 +6458,34 @@ private FlowAnalysisAnnotations GetParameterAnnotations(ParameterSymbol paramete
         {
             // Annotations are ignored when binding an attribute to avoid cycles. (Members used
             // in attributes are error scenarios, so missing warnings should not be important.)
-            return IsAnalyzingAttribute ? FlowAnalysisAnnotations.None : parameter.FlowAnalysisAnnotations;
+            if (IsAnalyzingAttribute)
+                return FlowAnalysisAnnotations.None;
+
+            // Conditional annotations are ignored on parameters of non-boolean members.
+            if (parameter.ContainingSymbol.GetTypeOrReturnType().Type.SpecialType != SpecialType.System_Boolean)
```

Comment:
It seems like we are just using information on the parameter symbol to make these adjustments, did you consider moving this logic into property FlowAnalysisAnnotations instead? Or possibly we wouldn't want to do these adjustment in all places this property is used?

Reply:
Yes, I considered it (for this PR and previously when the annotations were added). We had landed on keeping the symbol API straightfoward, it shows what annotations are there. That said, I don't feel too strongly either way.
@cston what do you think?

Reply:
Having the adjustment here seems fine to me, unless all callers should see the adjusted values.